coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -631,14 +633,59 @@ func (server *Server) CompleteJob(ctx context.Context, completed *proto.Complete
 			if err != nil {
 				return xerrors.Errorf("update workspace build: %w", err)
 			}
+
+			agentTimeouts := make(map[time.Duration]bool) // A set of agent timeouts.
 			// This could be a bulk insert to improve performance.
 			for _, protoResource := range jobType.WorkspaceBuild.Resources {
+				for _, protoAgent := range protoResource.Agents {
+					dur := time.Duration(protoAgent.GetConnectionTimeoutSeconds()) * time.Second
+					agentTimeouts[dur] = true
+				}
 				err = InsertWorkspaceResource(ctx, db, job.ID, workspaceBuild.Transition, protoResource, telemetrySnapshot)
 				if err != nil {
 					return xerrors.Errorf("insert provisioner job: %w", err)
 				}
 			}
 
+			// On start, we want to ensure that workspace agents timeout statuses
+			// are propagated. This method is simple and does not protect against
+			// notifying in edge cases like when a workspace is stopped soon
+			// after being started.
+			//
+			// Agent timeouts could be minutes apart, resulting in an unresponsive
+			// experience, so we'll notify after every unique timeout seconds.
```

Comment:
What "eveery unique timeout seconds" means?

Reply:
Let's say there are two agents with 120s timeout, then there will only be one unique timeout, 120s and we'll notify once.

Then in another scenario, we add a third agent with 240s timeout, there are now two unique timeouts `[120s, 240s]` and we will notify once at 120s, and another time at 240s.

Reply:
Ahh I see. Thanks for the explanation!